chore: Overhaul tracing documentation (#2052)
linkerd/website: linkerd.io/content/2-edge/tasks/distributed-tracing.md

To use distributed tracing, you'll need to:

- Install a trace collector and viewer on your cluster.

> - Update Linkerd with tracing enabled.

Author: SGTM

  Follow the [Installing Linkerd Guide](install/) if you haven't
  already done this.

## Install a trace collector

The first step of getting distributed tracing setup is installing a way to
collect, store, and view traces. The collector consumes spans emitted from the
mesh and your applications and sends them to the viewer which stores them and
serves a dashboard to view them.

Jaeger includes a collector, storage, and trace viewer in its all-in-one
installation, so we'll be using that for our examples.

To install Jaeger using Helm, first add the Jaeger Helm repository:

```bash
helm repo add jaegertracing https://jaegertracing.github.io/helm-charts
```

Then, install the Jaeger Helm chart:

```bash
helm install \
  --wait \
  --namespace jaeger-system \
  --create-namespace \
  --set allInOne.enabled=true \
  --set storage.type=memory \
  --set agent.enabled=false \
  --set collector.enabled=false \
  --set query.enabled=false \
  --set provisionDataStore.cassandra=false \
  jaeger jaegertracing/jaeger
```

> {{< note >}}

Reviewer: Sounds good to me.

The Jaeger all-in-one installation is very simple to set up and get running,
but it is not suitable for a production deployment. Determining what tracing
installation is suitable for your environment is beyond the scope of this page.
{{< /note >}}

## Update Linkerd with tracing enabled

There are a few values that need to be set on your Linkerd installation,
specifically to have proxies begin exporting traces as well as where to send
those traces. These values will point to the Jaeger collector installed

> above.

Reviewer 2: ```suggestion
There are a few values that need to be set on your Linkerd installation to enable exporting traces and to define where to send those traces. In this case, we'll configure Linkerd to send traces to the Jaeger collector we just installed.
```

(This started as me being nitpicky about the comma and then snowballed. 🤦‍♂️ 😂)

the source of your application. Tracing needs some way to tie incoming requests
to your application together with outgoing requests to dependent services. To do
this, some headers are added to each request that contain a unique ID for the
trace. Linkerd will propagatge both [w3c](https://www.w3.org/TR/trace-context/)
and [b3](https://github.com/openzipkin/b3-propagation) formats to tie these

> things together.

Reviewer 2: ```suggestion
trace. Linkerd will propagatge both [`w3c`](https://www.w3.org/TR/trace-context/)
and [`b3`](https://github.com/openzipkin/b3-propagation) formats to tie these
things together.

{{< note >}}

If both `w3c` and `b3` headers are present, Linkerd will propagate only the `w3c` headers.

{{< /note >}}
```

@sfleen, I vaguely remember that that's correct, but of course I'll defer to you here. 🙂

Author: That is correct, we always prefer w3c

the source side as the request traverses that proxy and two will be on the
destination side as the request is received by the remote proxy.

## Cleanup

> To cleanup, uninstall jaeger and emojivoto by running:

Reviewer 2: ```suggestion
To clean up, you'll uninstall emojivoto, turn Linkerd tracing off, and finally uninstall Jaeger:
```
